pkg/alerts/types.go

```diff
@@ -0,0 +1,161 @@
+package alerts
+
+// nolint:golint
+type AlertsIncidentPreference string
+
+const (
+	/* A condition will create a condition-level incident when it violates its critical threshold.
+	Other violating conditions will create their own incidents. */
+	PER_CONDITION AlertsIncidentPreference = "PER_CONDITION" // nolint:golint
+	/* Each target of each condition will create an entity-level incident upon critical violation.
+	Other violating targets even on the same condition will create their own incidents. */
+	PER_CONDITION_AND_TARGET AlertsIncidentPreference = "PER_CONDITION_AND_TARGET" // nolint:golint
+	/* A condition will create a policy-level incident when it violates its critical threshold.
+	Other violating conditions will be grouped into this incident. */
+	PER_POLICY AlertsIncidentPreference = "PER_POLICY" // nolint:golint
+)
+
+// nolint:golint
+type AlertsMutingRuleConditionGroupInput struct {
+	/* The individual MutingRuleConditions within the group. */
+	Conditions []AlertsMutingRuleConditionInput `json:"conditions"`
+
+	/* The operator used to combine all the MutingRuleConditions within the group. */
+	Operator AlertsMutingRuleConditionGroupOperator `json:"operator"`
+}
+
+// nolint:golint
+type AlertsMutingRuleConditionGroupOperator string
+
+const (
+	/* Match conditions by AND */
+	AND AlertsMutingRuleConditionGroupOperator = "AND" // nolint:golint
+	/* Match conditions by OR */
+	OR AlertsMutingRuleConditionGroupOperator = "OR" // nolint:golint
+)
+
+// nolint:golint
+type AlertsMutingRuleConditionInput struct {
+	/* The attribute on a violation. Expects one of:
+
+	* **accountId** - The account id
+	* **conditionId** - The alert condition id
+	* **policyId** - The alert policy id
+	* **policyName** - The alert policy name
+	* **conditionName** - The alert condition name
+	* **conditionType** - The alert condition type, such as app_metric
+	* **conditionRunbookUrl** - The alert condition's runbook url
+	* **product** - The target product (e.g., SYNTHETICS)
+	* **targetId** - The ID of the alerts target
+	* **targetName** - The name of the alerts target
+	* **nrqlEventType** - The NRQL event type
+	* **tag** - Arbitrary tags associated with some entity (e.g., FACET from a NRQL query)
+	* **nrqlQuery** - The NRQL query string */
```

Comment:
Do these show up in godocs?

Reply:
The probably don't as the first word in the comment is not the Name of the object.  We should refactor the typegen to resolve that as a follow up

Reply:
Let's find out.